### DISCUSSION TITLE
tls: failed to verify certificate: x509 with cluster openshift

### DISCUSSION THREAD

**Discussion by wintervargas** [argoproj/argo-cd#17067] [General]
Hello,

I am trying to create an application in argocd but I am having problems connecting to my openshift cluster through my domain certificate since it is self-signed.

This is the error:

Unable to create application: error while validating and normalizing app: error validating the repo: error getting k8s server version: Get "https://api.pre.ocp4.binter.cl:6443/version?timeout=32s": tls: failed to verify certificate: x509: certificate signed by unknown authority

I have this manifest for the integration with my openshift cluster:

apiVersion: v1
kind: Secret
metadata:
  namespace: argocd
  name: ocp-preprod
  labels:
    argocd.argoproj.io/secret-type: cluster
    environment: "preproductivo"
    provider: "redhat"
type: Opaque
stringData:
  name: ocp-preprod
  server: https://api.pre.ocp4.binter.cl:6443
  config: |
    {
      "token": "ey.....cm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2a",
      "insecure": "true"    }

Help me pleaseeee.

**Comment by jgwest** [comment]
`insecure` appears to be a string in your YAML, whereas it is a `bool` in the CR. You may wish to use the `argocd cluster add` CLI command to create your cluster Secret, if not already done.
